woocommerce/api/class-sv-wc-api-base.php

```diff
@@ -548,6 +571,79 @@ protected function get_request_duration() {
 	}
 
 
+	/**
+	 * Gets the request transient key for the current plugin and request data.
+	 *
+	 * Request transients can be disabled by using the filter below.
+	 *
+	 * @since 5.10.10
+	 *
+	 * @return string transient key
+	 */
+	protected function get_request_transient_key() : string {
+
+		// ex: wc_sv_requests_<md5 hash of plugin_id, request uri and request data>
+		return sprintf( 'wc_sv_requests_%s', md5( implode( '_', [
+			$this->get_plugin()->get_id(),
+			$this->get_request_uri(),
+			$this->get_request_body(),
+		] ) ) );
+	}
+
+
+	/**
+	 * Checks whether the current request is cacheable.
+	 *
+	 * @since 5.10.10
+	 *
+	 * @return bool
+	 */
+	protected function is_request_cacheable() : bool {
```

Comment:
For methods like these, which are checking or getting a value from the request, I would be more inclined to pass the request object in as an argument rather than always assume we want the request from `get_request()`. That would increase flexibility for APIs that have complex request logic. Granted at the point this is called `get_request()` _should_ return the only request we need to worry about, but again for flexibility if that's not the case in concrete plugins.

That also makes things more testable, should we add tests to for this class as we can test simple input and output and don't need to rely on mocking.

Reply:
Interesting, I was following the current pattern in this class, which is that we don't seem to pass around the request to any methods that deal with the request (ie, `get_request_uri`, `get_request_body`, etc).

Do you have any concrete plugins / use-cases in mind where the current implementation would be a problem? Or is it something we're looking to change for these other methods in the future as well?

Reply:
no specific use cases in mind, just a general preference that I've developed as we've worked on more modern projects. You're certainly right that the approach is consistent with existing handling. I doubt we'll go back and change things in this class any time in the near future. No big deal to leave as-is just for consistency.